Realm/RLMObjectSchema.mm

```diff
@@ -138,9 +138,17 @@ + (instancetype)schemaForObjectClass:(Class)objectClass {
     schema.accessorClass = objectClass;
     schema.unmanagedClass = objectClass;
     schema.isSwiftClass = isSwift;
-    schema.isEmbedded = [(id)objectClass isEmbedded];
     schema.hasCustomEventSerialization = [objectClass conformsToProtocol:@protocol(RLMCustomEventRepresentable)];
 
+    bool isEmbedded = [(id)objectClass isEmbedded];
+    bool isAsymmetric = [(id)objectClass isAsymmetric];
+    if (isEmbedded && isAsymmetric) {
+        @throw RLMException(@"Object '%@' is declared both as `Embedded` and `Asymmetric`, which is not supported", className);
```

Comment:
This should be an assertion, as it's not something that the user can do via API misuse.